src/01_getting_started/02_why_async.md

```diff
@@ -1,45 +1,121 @@
 # Why Async?
 
-We all love how Rust allows us to write fast, safe software. But why write
-asynchronous code?
+We all love how Rust allows us to write fast, safe software.
+But how does asynchronous programming fit into this vision?
 
-Asynchronous code allows us to run multiple tasks concurrently on the same OS
-thread. In a typical threaded application, if you wanted to download two
-different webpages at the same time, you would spread the work across two
-different threads, like this:
+Asynchronous programming, or async for short, is a _concurrent programming model_
+supported by an increasing number of programming languages.
+It lets you run a large number of concurrent
+tasks on a small number of OS threads, while preserving much of the
+look and feel of ordinary synchronous programming, through the
+`async/await` syntax.
+
+## Async vs other concurrency models
+
+Since the inception of concurrent programming, several models have been
+proposed with the intention of making concurrency simple and performant.
+Although details vary between different languages and frameworks,
+a brief overview of the main categories can help you understand how
+asynchronous programming fits within the field of concurrent programming:
+
+- **OS threads** don't require any changes to the programming model,
+  which makes them easier to
```

Comment:
I'm surprised to hear you say this; it can be pretty difficult to reason about thread safety and to test threaded code (although Rust does make it easier).

Reply:
Ah yes, this is because _expressing concurrency_ is easy, but synchronization is hard (or super-human-hard, depending on who you ask). Clarified.